File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityCarsTest.cpp
#include "ActivityCarsTest.h"
#include <memory>
#include "Cars/Car.h"
#include <alf/library/BasicInputOutput.hpp>
#include <string>
#include "Cars/Vehicle.h"

void ActivityCarsTest::CarsTest()
{
  std::shared_ptr<Cars::Car> car = std::make_shared<Cars::Car>();
  car->startEngine();
  alf::library::BasicInputOutput::WriteLine(std::to_string(car->distance));
  std::shared_ptr<Cars::Vehicle> vehicle = car;
  vehicle->drive();
  alf::library::BasicInputOutput::WriteLine(std::to_string(vehicle->distance));
  alf::library::BasicInputOutput::WriteLine(std::to_string(car->distance));
  car->drive();
  alf::library::BasicInputOutput::WriteLine(std::to_string(car->distance));
  alf::library::BasicInputOutput::WriteLine(car->getColor());
  car->setColor(std::string("Red"));
  alf::library::BasicInputOutput::WriteLine(car->getColor());

  // nested classifiers
  Cars::Car::Model model = { std::string("My Model"), Cars::Car::ModelKind::Limousine };
  car->model = model;
  alf::library::BasicInputOutput::WriteLine(car->model.Name);
  if (model.Kind == Cars::Car::ModelKind::Limousine)
  {
    alf::library::BasicInputOutput::WriteLine(std::string("Limousine"));
  }

  std::shared_ptr<Cars::Car::Wheel> wheel = std::make_shared<Cars::Car::Wheel>();
  car->addWheel(wheel);
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityCarsTest.cs
/// <filename>
///     ActivityCarsTest.cs
/// </filename>
using Cars;
using Alf.Library;
using System;

public class ActivityCarsTest
{
    public static void Execute()
    {
        Cars.Car car = new Cars.Car();
        car.startEngine();
        BasicInputOutput.WriteLine(Convert.ToString(car.distance));
        Cars.Vehicle vehicle = car;
        vehicle.drive();
        BasicInputOutput.WriteLine(Convert.ToString(vehicle.distance));
        BasicInputOutput.WriteLine(Convert.ToString(car.distance));
        car.drive();
        BasicInputOutput.WriteLine(Convert.ToString(car.distance));
        BasicInputOutput.WriteLine(car.getColor());
        car.setColor("Red");
        BasicInputOutput.WriteLine(car.getColor());

        // nested classifiers
        Cars.Car.Model model = new Cars.Car.Model("My Model", Car.ModelKind.Limousine);
        car.model = model;
        BasicInputOutput.WriteLine(car.model.Name);
        if (model.Kind == Car.ModelKind.Limousine)
        {
            BasicInputOutput.WriteLine("Limousine");
        }

        Cars.Car.Wheel wheel = new Cars.Car.Wheel();
        car.addWheel(wheel);
    }

    public static void Main(string[] args)
    {
        ActivityCarsTest.Execute();
    }
}
